index.js

```diff
@@ -448,6 +449,43 @@ module.exports = function(options) {
     return self.callAllAndEmit('afterInit', 'afterInit', callback);
   }
 
+  function lintModules(callback) {
+    _.each(self.modules, function(module, name) {
+      if (name.match(/-widgets$/) && (!extending(module)) && (!module.options.fromScratch)) {
+        lint('The module ' + name + ' does not extend anything.\n\nA widgets module usually extends `apostrophe-widgets` or\n`apostrophe-pieces-widgets`. Or possibly you forgot to npm install something.\n\nIf you are sure you are doing the right thing, set the `fromScratch` option\nto `true` for this module.');
+      } else if (name.match(/-pages$/) && (name !== 'apostrophe-pages') && (!extending(module)) && (!module.options.fromScratch)) {
+        lint('The module ' + name + ' does not extend anything.\n\nA -pages module usually extends `apostrophe-custom-pages` or\n`apostrophe-pieces-pages`. Or possibly you forgot to npm install something.\n\nIf you are sure you are doing the right thing, set the `fromScratch` option\nto `true` for this module.');
+      } else if ((!extending(module)) && (!hasConstruct(name)) && (module.options.hasCode !== false)) {
+        lint('The module ' + name + ' does not extend anything and does not have a\n`construct` function. This usually means that you:\n\n1. Forgot `extend`\n2. Configured a module that comes from npm without npm installing it\n3. Just haven\'t written your \'index.js\` yet\n\nIf you really want a module with no code, set the `hasCode` option\nexplicitly to `false` for this module.');
```

Comment:
- "Forgot `extend`" isn't very clear. Maybe "Forgot to `extend` another module."
- "without npm installing" is clunky. Maybe "without actually installing."
- This is a smaller thing, but I have an aversion to "just" since it can diminish a thing ("Just haven't written your 'index.js` yet"). "Simply" feels better, or it could be removed completely.